src/nvim/tui/tui.c

```diff
@@ -929,12 +932,25 @@ static void tui_set_mode(UI *ui, ModeShape mode)
   TUIData *data = ui->data;
   cursorentry_T c = data->cursor_shapes[mode];
 
-  if (c.id != 0 && ui->rgb) {
-    int attr = syn_id2attr(c.id);
-    if (attr > 0) {
-      HlAttrs *aep = syn_cterm_attr2entry(attr);
-      UNIBI_SET_NUM_VAR(data->params[0], aep->rgb_bg_color);
-      unibi_out_ext(ui, data->unibi_ext.set_cursor_color);
+  if (ui->rgb) {
```

Comment:
This form avoids some nesting and diff churn:

```c
if (c.id != 0 && ui->rgb) {
  int attr = syn_id2attr(c.id);
  if (attr > 0) {
    HlAttrs *aep = syn_cterm_attr2entry(attr);
    if (aep->rgb_ae_attr & HL_INVERSE) {
      // We reset the cursor color to the default because AFAIK it is
      // impossible to directly set cursor color to inverse.  Hopefully the
      // user's default cursor color is inverse.
      unibi_out_ext(ui, data->unibi_ext.reset_cursor_color);
    } else {
      UNIBI_SET_NUM_VAR(data->params[0], aep->rgb_bg_color);
      unibi_out_ext(ui, data->unibi_ext.set_cursor_color);
    }
  }
} else if (c.id == 0) {
  // Reset the cursor color to default if the user does not specify cursor
  // color.
  unibi_out_ext(ui, data->unibi_ext.reset_cursor_color);
}
```